### PR TITLE
Migrated Shift app to CBV

### PR DESCRIPTION
**Pull request by amruthasangeeth**
Migrated function based views in Shift app to CBV. The following are the the CBVs used for migration:
DeleteView- To delete a Shift
FormView- To display the forms in Shift app(like AddHours, EditHours etc)
ListView- To display the list of shifts and jobs assigned.
TemplateView- To render a template
UpdateView- To edit the Shift object

### CODE REVIEW

**Comment by tapasweni-pathak** [comment]
@amruthasangeeth When I open your pull request, all but one file are coming in diff. Diff should show only relevant changes, you have to make sure that is the case. I should not compare line by line manually, when there is something which ideally should do it for me. Please make sure this in future pull requests and update this pull request.

**Comment by tapasweni-pathak** [comment]
@amruthasangeeth Point me to the file which you are not able to fix.

**Comment by amruthasangeeth** [comment]
@tapasweni-pathak Now just the diff are shown in Files Changed. Removed the issue of showing the entire file as the diff.

File: vms/shift/views.py [hunk 1]
```diff
@@ -5,144 +5,140 @@
...
+                    add_shift_hours(
+                        volunteer_id,
+                        shift_id,
+                        start_time,
+                        end_time
+                    )
+                    return HttpResponseRedirect(reverse(
+                        'shift:view_hours',
+                        args=(volunteer_id,)
+                    ))
```
Comment (tapasweni-pathak) [inline vms/shift/views.py]: @amruthasangeeth Did you run pep8 over these changes?

File: vms/shift/views.py [hunk 2]
```diff
@@ -759,54 +609,37 @@ def view_volunteer_shifts(request, volunteer_id):
...
-                        organization
-                        )
-                return render(
-                    request,
-                    'shift/volunteer_search.html',
-                    {'form': form, 'has_searched': True, 'volunteer_list': volunteer_list}
-                    )
-        else:
-            form = SearchVolunteerForm()
-            volunteer_list = get_all_volunteers()
```
Comment (tapasweni-pathak) [inline vms/shift/views.py]: @amruthasangeeth Did you handle this case in the new code?

Comment (amruthasangeeth) [inline vms/shift/views.py]: @tapasweni-pathak That is already handled by FormView. It shows "Your search return no results!" when the form entered is invalid.

**Comment by tapasweni-pathak** [comment]
#288.

**Comment by tapasweni-pathak** [comment]
Thank you. :sparkle:

**Pull request merged**